lab_3_ann_retriever/main.py

```diff
@@ -147,49 +147,99 @@ def __init__(self, corpus: list[list[str]]) -> None:
         Initialize an instance of the Vectorizer class.
 
         Args:
-            corpus (list[list[str]]): Tokenized documents to vectorize
+            corpus (list[list[str]]): Tokenized documents to vectorize.
         """
+        self._corpus = corpus
+        self._vocabulary = []
+        self._idf_values = {}
+        self._token2ind = {}
 
-    def build(self) -> bool:
+    def build(self) -> None:
         """
-        Build vocabulary with tokenized_documents.
-
-        Returns:
-            bool: True if built successfully, False in other case
+        Builds vocabulary with tokenized_documents.
         """
+        if not self._corpus:
+            return None
+
+        self._vocabulary = sorted(list({term for doc in self._corpus for term in doc}))
+        self._idf_values = calculate_idf(self._vocabulary, self._corpus)
+        self._token2ind = {term: index for index, term in enumerate(self._vocabulary)}
+        if not self._idf_values or not self._vocabulary or not self._token2ind:
+            raise ValueError
+
+        return None
 
     def vectorize(self, tokenized_document: list[str]) -> Vector | None:
         """
         Create a vector for tokenized document.
 
         Args:
-            tokenized_document (list[str]): Tokenized document to vectorize
+            tokenized_document (list[str]): Tokenized document to vectorize.
 
         Returns:
-            Vector | None: TF-IDF vector for document
+            Vector | None: TF-IDF vector for document.
 
         In case of corrupt input arguments, None is returned.
         """
+#       vector = tuple(float, ...)
+
+        if not isinstance(tokenized_document, list) or not tokenized_document \
+                or not all(isinstance(term, str) for term in tokenized_document):
+            return None
+
+        return self._calculate_tf_idf(tokenized_document)
+
+    def _calculate_tf_idf(self, document: list[str]) -> Vector | None:
```

Comment:
why is this function is fully new?

Reply:
i dont know..

Reply:
unfortunately, the review last week is on Monday and Wednesday

Reply:
fixed